x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileDomainIntegTests.java

```diff
@@ -249,14 +259,137 @@ public void testGetProfileByAuthenticationWillNotCheckRealmNameForFileOrNativeRe
         assertThat(future1.actionGet().uid(), equalTo(profile1.uid()));
     }
 
+    public void testConcurrentCreationOfNewProfiles() throws InterruptedException {
```

Comment:
Do you think a similar integ test for an already existing profile would be worthwhile?

Reply:
I mean test activating profiles concurrently when there are documents with a "collision" doc id.

Reply:
Because it is important that retrying to create the profile doc uses a deterministic new id as well (not just the first id is deterministic).

Reply:
Good point. I add a randomization to add an existing collision doc.